Lib/_pyrepl/fancycompleter.py

```diff
@@ -0,0 +1,175 @@
+            delims = readline.get_completer_delims()
+            delims = delims.replace('[', '')
+            delims = delims.replace(']', '')
+            readline.set_completer_delims(delims)
+
+    def complete(self, text, state):
+        # if you press <tab> at the beginning of a line, insert an actual
+        # \t. Else, trigger completion.
+        if text == "":
+            return ('\t', None)[state]
+        else:
+            return rlcompleter.Completer.complete(self, text, state)
+
+    def _callable_postfix(self, val, word):
+        # disable automatic insertion of '(' for global callables
+        return word
+
+    def global_matches(self, text):
+        names = rlcompleter.Completer.global_matches(self, text)
+        prefix = commonprefix(names)
+        if prefix and prefix != text:
+            return [prefix]
+
+        names.sort()
+        values = []
+        for name in names:
+            clean_name = name.rstrip(': ')
+            if clean_name in keyword.kwlist:
+                values.append(None)
+            else:
+                try:
+                    values.append(eval(name, self.namespace))
+                except Exception as exc:
+                    values.append(None)
+        if self.use_colors and names:
+            return self.colorize_matches(names, values)
+        return names
+
+    def attr_matches(self, text):
+        expr, attr = text.rsplit('.', 1)
+        if '(' in expr or ')' in expr:  # don't call functions
+            return []
+        try:
+            thisobject = eval(expr, self.namespace)
+        except Exception:
+            return []
+
+        # get the content of the object, except __builtins__
+        words = set(dir(thisobject)) - {'__builtins__'}
+
+        if hasattr(thisobject, '__class__'):
+            words.add('__class__')
+            words.update(rlcompleter.get_class_members(thisobject.__class__))
+        names = []
+        values = []
+        n = len(attr)
+        if attr == '':
+            noprefix = '_'
+        elif attr == '_':
+            noprefix = '__'
+        else:
+            noprefix = None
+        words = sorted(words)
+        while True:
+            for word in words:
+                if (
+                    word[:n] == attr
+                    and not (noprefix and word[:n+1] == noprefix)
+                ):
+                    try:
+                        val = getattr(thisobject, word)
+                    except Exception:
+                        val = None  # Include even if attribute not set
+
+                    names.append(word)
+                    values.append(val)
+            if names or not noprefix:
+                break
+            if noprefix == '_':
+                noprefix = '__'
+            else:
+                noprefix = None
+
+        if not names:
+            return []
+
+        if len(names) == 1:
+            return [f'{expr}.{names[0]}']  # only option, no coloring.
```

Comment:
Should we color anyway for consistency?

Reply:
no. When you return a single completion, readline inserts it in the prompt. If we return colors here, we would be ANSI codes in the prompt, which is surely not what you want

Reply:
Agreed, changing the colors in the prompt is not what we want. Could you update the comment with the reason?